src/NServiceBus.RabbitMQ/Routing/DefaultRoutingKeyConvention.cs

```diff
@@ -23,6 +24,16 @@ static string GetRoutingKey(Type type, string key = "")
 
             var implementedInterface = interfaces.FirstOrDefault();
 
+            if (interfaces.Count > 0)
+            {
+                Logger.Warn("The direct routing topology might not deliver message types with interfaces in their inheritance hierarchy. Message types used for publishing should not implement interfaces.");
```

Comment:
Maybe we need to take a step back and reevaluate what we're trying to accomplish with this warning instead.

If we've detected that there is just a single interface, then we don't actually have a problem, because it should work when there is just one.

The problem is actually when we have more than one, which is what the Error log entry is pointing out.

Maybe we just need to remove this `Count > 0` warning, and only log in the `Count > 1` error case?